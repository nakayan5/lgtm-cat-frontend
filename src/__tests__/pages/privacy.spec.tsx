/**
 * @jest-environment jsdom
 */
import fs from 'fs';

import { render } from '@testing-library/react';
import React from 'react';

import PrivacyPage from '../../pages/privacy';

test('PrivacyPage Snapshot test', async () => {
  const fsPromise = fs.promises;

  const privacy = await fsPromise.readFile(
    `${process.cwd()}/src/docs/privacy.md`,
    {
      encoding: 'utf8',
    },
  );

  const { asFragment } = render(<PrivacyPage privacy={privacy} />);
  expect(asFragment()).toMatchSnapshot();
});
