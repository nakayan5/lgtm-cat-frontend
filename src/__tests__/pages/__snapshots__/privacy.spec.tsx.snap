// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrivacyPage Snapshot test 1`] = `
<DocumentFragment>
  <div
    class="hero is-fullheight"
  >
    <header>
      <nav
        class="navbar navbar-padding"
      >
        <div
          class="container"
          style="display: block;"
        >
          <div
            class="navbar-brand"
          >
            <a
              class="navbar-item"
              href="/"
            >
              <p
                class="is-size-4 has-text-black"
              >
                LGTMeow
              </p>
            </a>
            <div
              style="align-items: center; display: flex; margin-left: auto;"
            />
          </div>
        </div>
      </nav>
    </header>
    <div
      class="container"
      style="display: block;"
    >
      <div
        class="content mb-6 mx-3"
      >
        <h1>
          プライバシーポリシー
        </h1>
        <p>
          nekochans（以下，「運営チーム」といいます。）は，本ウェブサイト上で提供するサービス（以下,「本サービス」といいます。）における，ユーザーの個人情報の取扱いについて，以下のとおりプライバシーポリシー（以下，「本ポリシー」といいます。）を定めます。
        </p>
        <h2>
          1. 個人情報の定義
        </h2>
        <p>
          「個人情報」とは，個人情報保護法にいう「個人情報」を指すものとし，生存する個人に関する情報であって，当該情報に含まれる氏名，生年月日，住所，電話番号，連絡先その他の記述等により特定の個人を識別できる情報及び容貌，指紋，声紋にかかるデータ，及び健康保険証の保険者番号などの当該情報単体から特定の個人を識別できる情報（個人識別情報）を指します。
        </p>
        <h2>
          2. 個人情報の取得方法
        </h2>
        <p>
          本サービスでは登録およびご利用に際して以下の情報を取得し，運営チームはそれらを個人情報として取り扱います。
        </p>
        <ul>
          <li>
            GitHub アカウントに関する情報
          </li>
          <li>
            メールアドレス
          </li>
        </ul>
        <h2>
          3. 個人情報を取得・利用する目的
        </h2>
        <p>
          運営チームが個人情報を取得・利用する目的は，以下のとおりです。
        </p>
        <ol>
          <li>
            本サービスの提供・運営のため
          </li>
          <li>
            ユーザーからのお問い合わせに回答するため
          </li>
          <li>
            利用規約に違反したユーザーや，不正・不当な目的でサービスを利用しようとするユーザーの特定をし，ご利用をお断りするため
          </li>
          <li>
            ユーザーにご自身の登録情報の閲覧や変更，削除，ご利用状況の閲覧を行っていただくため
          </li>
          <li>
            本サービス全体の統計情報としてデータを解析するため
          </li>
          <li>
            上記の利用目的に付随する目的
          </li>
        </ol>
        <h2>
          4. 個人情報を取得・利用する目的
        </h2>
        <p>
          運営チームは，利用目的が変更前と関連性を有すると合理的に認められる場合に限り，個人情報の利用目的を変更するものとします。
利用目的の変更を行った場合には，変更後の目的について，運営チーム所定の方法により，ユーザーに通知し，または本ウェブサイト上に公表するものとします。
        </p>
        <h2>
          5. 個人情報の第三者提供
        </h2>
        <ol>
          <li>
            <p>
              運営チームは，次に掲げる場合を除いて，あらかじめユーザーの同意を得ることなく，第三者に個人情報を提供することはありません。ただし，個人情報保護法その他の法令で認められる場合を除きます。
            </p>
            <ol>
              <li>
                人の生命，身体または財産の保護のために必要がある場合であって，本人の同意を得ることが困難であるとき
              </li>
              <li>
                公衆衛生の向上または児童の健全な育成の推進のために特に必要がある場合であって，本人の同意を得ることが困難であるとき
              </li>
              <li>
                国の機関もしくは地方公共団体またはその委託を受けた者が法令の定める事務を遂行することに対して協力する必要がある場合であって，本人の同意を得ることにより当該事務の遂行に支障を及ぼすおそれがあるとき
              </li>
              <li>
                予め次の事項を告知あるいは公表し，かつ運営チームが個人情報保護委員会に届出をしたとき
                <ol>
                  <li>
                    利用目的に第三者への提供を含むこと
                  </li>
                  <li>
                    第三者に提供されるデータの項目
                  </li>
                  <li>
                    第三者への提供の手段または方法
                  </li>
                  <li>
                    本人の求めに応じて個人情報の第三者への提供を停止すること
                  </li>
                  <li>
                    本人の求めを受け付ける方法
                  </li>
                </ol>
              </li>
            </ol>
          </li>
          <li>
            <p>
              前項の定めにかかわらず，次に掲げる場合には，当該情報の提供先は第三者に該当しないものとします。
            </p>
            <ol>
              <li>
                運営チームが利用目的の達成に必要な範囲内において個人情報の取扱いの全部または一部を委託する場合
              </li>
              <li>
                合併その他の事由による事業の承継に伴って個人情報が提供される場合
              </li>
            </ol>
          </li>
        </ol>
        <h2>
          6. Cookie 及び LocalStorage の利用について
        </h2>
        <p>
          本サービスではログイン状態の確認，ユーザー利便性の向上，統計データの取得のため，Cookie 及び LocalStorage を利用しています。Cookie 及び Localstorage の設定を変更し，機能を無効にすることができます。但し，その結果，本サービスの全部または一部の機能がご利用いただけなくなることがあります。
        </p>
        <h2>
          7. アクセス解析ツールについて
        </h2>
        <p>
          本サービスでは，Google によるアクセス解析ツール「Google Analytics」を利用しています。この Google Analytics はトラフィックデータの収集のために Cookie を使用しています。このトラフィックデータは匿名で収集されており，個人を特定するものではありません。Google Analytics から提供される Cookie の使用に関する説明，Cookie によって収集される情報について詳しくお知りになりたい方は，Google Analytics のプライバシーポリシーをご確認ください。
この機能は Cookie を無効にすることで収集を拒否することが出来ますので，お使いのブラウザの設定をご確認ください。
        </p>
        <h2>
          8. プライバシーポリシーの変更
        </h2>
        <p>
          本ポリシーの内容は，法令その他本ポリシーに別段の定めのある事項を除いて，ユーザーに通知することなく，変更することができるものとします。
運営チームが別途定める場合を除いて，変更後のプライバシーポリシーは，本ウェブサイトに掲載したときから効力を生じるものとします。
        </p>
        <h2>
          9. お問い合わせ窓口
        </h2>
        <p>
          本ポリシーに関するお問い合わせは，下記の窓口までお願いいたします。
        </p>
        <p>
          <a
            href="https://docs.google.com/forms/d/e/1FAIpQLSf0-A1ysrWQFCDuOZY8f2uH5KhUCB5yqi7TlLEsgl95Q9WKtw/viewform"
          >
            お問い合わせフォーム
          </a>
        </p>
        <p>
          2021 年 2 月 22 日 制定
        </p>
      </div>
    </div>
    <footer
      class="footer"
    >
      <div
        class="container"
      >
        <nav
          class="level"
        >
          <div
            class="level-left"
          >
            <div
              class="level-item breadcrumb"
            >
              <ul>
                <li>
                  <a
                    href="/terms"
                  >
                    利用規約
                  </a>
                </li>
                <li>
                  <a
                    href="/privacy"
                  >
                    プライバシーポリシー
                  </a>
                </li>
              </ul>
            </div>
          </div>
          <div
            class="level-right"
          >
            <div
              class="content has-text-centered"
            >
              <p>
                Copyright (c) nekochans
              </p>
            </div>
          </div>
        </nav>
      </div>
    </footer>
  </div>
</DocumentFragment>
`;
